### PR TITLE
Add frontend (JSP, CSS & JS) files for homepage

### PR DESCRIPTION
**Pull request by lisasgoh**

### CODE REVIEW

File: src/main/webapp/WEB-INF/jsp/home.jsp [hunk 1]
```diff
@@ -19,17 +20,44 @@
...
+      <!-- Search Bar-->
+      <form class="form-inline active-pink-3 active-pink-4">
+        <i class="fas fa-search" aria-hidden="true"></i>
+        <input class="form-control form-control-sm ml-3 w-75" type="text" placeholder="Search" aria-label="Search">
+      </form>
+    </div>
+  </header>
+  <section class="page-section" id="popular-deals">
+    <h2 class="page-section-heading text-center text-uppercase text-secondary mb-0">Trending Deals</h2>
+    <%@include file="/WEB-INF/components/section-carousel.html"%>
```
Comment (limli) [inline src/main/webapp/WEB-INF/jsp/home.jsp:42]: try to code the hardcoded data in a way that can be easily updated when the backend is made

File: src/main/webapp/js/home-page.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    },
+  ],
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+  createCarousel();
+  createHomePage(homePage);
+});
+
+function createHomePage(homePage) {
```
Comment (limli) [inline src/main/webapp/js/home-page.js:66]: add jsdocs

File: src/main/webapp/WEB-INF/jsp/home.jsp [hunk 3]
```diff
@@ -19,17 +20,44 @@
...
+  </section>
   <%@include file="/WEB-INF/components/footer.html"%>
   <!-- Bootstrap core JS-->
   <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
   <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.bundle.min.js"></script>
   <!-- Third party plugin JS-->
   <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-easing/1.4.1/jquery.easing.min.js"></script>
+  <!-- Contact form JS-->
+  <script src="assets/mail/jqBootstrapValidation.js"></script>
```
Comment (limli) [inline src/main/webapp/WEB-INF/jsp/home.jsp]: don't need these js files

**Comment by SuyashLakhotia** [comment]
@lisasgoh Please resolve the comments by @limli and then we'll take a look at your PR.

File: src/main/webapp/js/home-page.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      const indicatorListChild = document.createElement('li');
+      indicatorListChild.dataset.target = '#carousel-' +i;
+      indicatorListChild.setAttribute('data-slide-to', j);
+      const carouselItemListChild = document.createElement('div');
+      carouselItemListChild.classList.add('carousel-item');
+      const rowElement = document.createElement('div');
+      rowElement.className='row';
+      const numCol = 'col-md-' + 12/numDealPerSlide;
+      for (let k = 0; k < numDealPerSlide; k++) {
+        rowElement.innerHTML += '<div class="' +
```
Comment (limli) [inline src/main/webapp/js/home-page.js]: change all the string + ... + ... to use template strings will look better
```
rowElement.innerHTML += `
  <div class="${numCol} mt-5">
    <div id=deal-card-${i} class="card deal-card h-100">
    <img class="card-img-top home-deal-img" src="" alt="">
    <div class="card-body d-flex flex-column">
    <h5 class="card-title deal-title"></h5>
    <p class="card-text deal-text"></p>
    <p class="card-text deal-poster"></p>
    <a href="#"
        class="btn btn-primary align-self-end mt-auto float-right">
      See More
    </a>
  </div>`;
```

File: src/main/webapp/WEB-INF/jsp/home.jsp [hunk 5]
```diff
@@ -9,6 +9,13 @@
   <title>Freelancer - Start Bootstrap Theme</title>
```
Comment (limli) [inline src/main/webapp/WEB-INF/jsp/home.jsp]: maybe change title to something like DealFinder

File: src/main/webapp/js/home-page.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+          numCol + ' mt-5">' +
+          '<div ' + 'id=deal-card-' + i + ' class="card deal-card h-100">' +
+          '<img class="card-img-top home-deal-img" src="" alt="">' +
+          '<div class="card-body d-flex flex-column">' +
+          '<h5 class="card-title deal-title"></h5>' +
+          '<p class="card-text deal-text"></p>' +
+          '<p class="card-text deal-poster"></p>' +
+          '<a href="#" class="btn btn-primary align-self-end ' +
+          'mt-auto float-right">See More</a>' +
```
Comment (limli) [inline src/main/webapp/js/home-page.js]: maybe can hide the "See More" if there is nothing there

File: src/main/webapp/js/home-page.js (hunk 4 above)
Comment (limli) [reply src/main/webapp/js/home-page.js]: you rowElement had a `</div></div>` at the end, im not sure what it is for

File: src/main/webapp/js/home-page.js (hunk 6 above)
Comment (lisasgoh) [reply src/main/webapp/js/home-page.js]: its to view the individual deal pages though

Comment (limli) [reply src/main/webapp/js/home-page.js]: like if there is only 1 deal on the page but the carousel have 4 items, i think the "See More" for the 2nd to 4th item on the carousel shouldn't be visible

Comment (limli) [reply src/main/webapp/js/home-page.js]: Screenshot 2020-07-06 at 4 38 11 PM

like the white blocks shouldn't be visible

File: src/main/webapp/WEB-INF/components/section-row.html [hunk 7]
```diff
@@ -0,0 +1,46 @@
+<div class="row">
```
Comment (limli) [inline src/main/webapp/WEB-INF/components/section-row.html]: is this file used anywhere now?

Comment (SuyashLakhotia) [reply src/main/webapp/WEB-INF/components/section-row.html]: +1, can this be removed now?

File: src/main/webapp/WEB-INF/components/section-row.html [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        <div class="card deal-card h-100">
+            <img class="card-img-top home-deal-img" src="" alt="">
+            <div class="card-body d-flex flex-column">
+            <h5 class="card-title deal-title"></h5>
+            <p class="card-text deal-text"></p>
+            <p class="card-text deal-poster"></p>
+            <a href="#" class="btn btn-primary align-self-end mt-auto float-right">See More</a>
+            </div>
+        </div>
+    </div>
```
Comment (SuyashLakhotia) [inline src/main/webapp/WEB-INF/components/section-row.html]: Seems like each of these blocks are the same? Consider putting in a separate file and just repeating here using include?

File: src/main/webapp/css/styles.css [hunk 9]
```diff
@@ -10277,7 +10277,32 @@ body > .container {
   flex-grow: 1;
 }
 
+.header-bg-img {
+  background-image: url(/../assets/img/food-photo.jpeg);
```
Comment (SuyashLakhotia) [inline src/main/webapp/css/styles.css]:
```suggestion
  background-image: url('/../assets/img/food-photo.jpeg');
```

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/components/section-row.html (hunk 8 above)
Comment (lisasgoh) [reply src/main/webapp/WEB-INF/components/section-row.html]: I deleted the html file instead cos im creating the deal card elements in the home-page.js now

**Review by SuyashLakhotia — DISMISSED** [review]

File: src/main/webapp/js/home-page.js [hunk 10]
```diff
@@ -59,10 +59,21 @@ const homePage = {
...
+  const oldonload = window.onload;
+  if (typeof oldonload == 'function') {
+    window.onload = function() {
+      oldonload();
+      func();
+    };
+  } else {
+    window.onload = func;
+  }
+}
```
Comment (SuyashLakhotia) [inline src/main/webapp/js/home-page.js]: Use the function from https://github.com/googleinterns/step203-2020/blob/master/src/main/webapp/js/util.js?

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

**Review by limli — APPROVED** [review]

**Review by SuyashLakhotia — APPROVED** [review]

**Pull request merged**